src/nyc_trees/libs/auth_backend.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+from django.contrib.auth.backends import ModelBackend
+
+from apps.core.models import User
+
+
+class EmailOrUsernameAuthentication(ModelBackend):
+    def authenticate(self, username=None, password=None):
+        user = None
+        try:
```

Comment:
I've pushed a commit which should address this by adding some extra validation to prevent this cornercase from happening.